include/seqan3/range/container/constexpr_string.hpp

```diff
@@ -0,0 +1,430 @@
+        for (const_iterator it = cbegin(); it != cend(); ++it, ++it_rhs)
+        {
+            if (*it != *it_rhs)
+                return false;
+        }
+
+        return true;
+    }
+
+    /*!\brief Compares two strings for inequality.
+     *
+     * \returns `bool` `false` if both container are equal, `true` otherwise.
+     */
+    template <std::size_t N2>
+    constexpr bool operator!=(constexpr_string<N2> const & rhs) const noexcept
+    {
+        return !(*this == rhs);
+    }
+
+    /*!\brief Swaps two strings.
+     * Note both strings must have the same size in order to swap them.
+     */
+    constexpr void swap(constexpr_string & other) noexcept
+    {
+        _lit.swap(other._lit);
+    }
+
+    /*!\brief Returns the content represented as std::string.
+     *
+     * \returns \c std::string The stored string.
+     * \par Exceptions
+     * Strong exception guarantee. No data is modified.
+     */
+    std::string string() const
+    {
+        return std::string{ _lit.cbegin(), _lit.cend() - 1};
+    }
+
+    /*!\brief Returns the content represented as 0-terminated c-style string.
+     *
+     * \returns `const char *` The stored string.
+     * \par Exceptions
+     * No-throw guarantee.
+     */
+    constexpr const char * c_str() const noexcept
+    {
+        return _lit.data();
+    }
+    //!\}
+
+    //!\brief Implicit string conversion.
+    operator std::string() const
+    {
+        return this->string();
+    }
+};
+
+/*!\brief Swaps two constexpr_strings.
+ *
+ * \par Excecptions
+ * noexcept if `lhs.swap(rhs)` is `noexcept`. Otherwise it holds strong exception guarantee.
+ */
+template <std::size_t N>
+void swap(constexpr_string<N> & lhs, constexpr_string<N> & rhs) noexcept(noexcept(lhs.swap(rhs)))
+{
+    lhs.swap(rhs);
+}
+
+/*!\name Deduction guides
+ * \{
+ */
+//!\brief Deduces constexpr_string from string literals.
+template <std::size_t N>
+constexpr_string(const char (&)[N]) -> constexpr_string<N - 1>;
+
+//!\brief Deduces constexpr_string from std::array of type char.
+template <std::size_t N>
+constexpr_string(std::array<char, N> const &) -> constexpr_string<N>;
+
+//!\brief Deduces constexpr_string from char.
+constexpr_string(const char) -> constexpr_string<1>;
+//!\}
+
+/*!\brief Helper function to construct the seqan3::detail::constexpr_string from c-style string.
+ * \tparam Is Parameter pack over all indices of extractable characters.
+ * \param c_str The c-style string to extract characters from.
+ * \see make_constexpr_string
+ * \return An seqan3::detail::constexpr_string of size `sizeof...(Is)` containing the respective part of `c_str`.
+ * \par Exceptions
+ * No-throw guarantee.
+ * \par Thread-Safety
+ * Thread safe.
+ * \ingroup container
+ */
+template <std::size_t ...Is>
+constexpr auto
+make_constexpr_string_impl(const char * const c_str, std::index_sequence<Is...> const & /*index_seq*/) noexcept
+{
+    return constexpr_string{ std::array{c_str[Is]...} };
+}
```

Comment:
does this have to be a separate function, can you not add this to the actual function as local function or lambda?
If not, please move it to detail.